pallets/parachain-staking/src/lib.rs

```diff
@@ -507,13 +737,19 @@ pub mod pallet {
 
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
-		// This upgrade fixes a bug that may have led to an incorrect `Total` (total staked)
 		fn on_runtime_upgrade() -> Weight {
+			// migrate from Collator -> Collator2
+			for (acc, collator_state) in CollatorState::<T>::iter() {
```

Comment:
```suggestion
			for (acc, collator_state) in CollatorState::<T>::drain() {
```
You want to take them out of the old storage location right?